### PR TITLE
[ROCm] Cap SDPA head dim at 256 on gfx11xx for AOTriton 0.11 and later

### LINKED ISSUE #189849: Flash SDPA wrong output for head_dim=512 on gfx1151 (ROCm 7.15 regression)
### 🐛 Describe the bug

We observed corrupted output images when running a Stable Diffusion 3 pipeline on AMD gfx1151. After isolating the issue, we identified that scaled_dot_product_attention with head_dim=512 produces numerically incorrect results on affected builds — output values saturate to exactly 512.0 instead of the expected < 1.0 range.

Note: This issue is specific to `gfx1151`. The same PyTorch builds were tested on `gfx1201` and do not exhibit this problem — head_dim=512 produces correct results on gfx1201 across all tested versions.

### Regression test
```
[X] pip install --index-url https://rocm.nightlies.amd.com/whl-multi-arch/ "torch[device-gfx1151]"==2.14.0a0+rocm7.15.0a20260713
[X] pip install --index-url https://rocm.nightlies.amd.com/whl-multi-arch/ "torch[device-gfx1151]"==2.14.0a0+rocm7.15.0a20260711
[X] pip install --index-url https://rocm.nightlies.amd.com/whl-multi-arch/ "torch[device-gfx1151]"==2.14.0a0+rocm7.15.0a20260630
[X] pip install --index-url https://rocm.nightlies.amd.com/whl-multi-arch/ "torch[device-gfx1151]"==2.14.0a0+rocm7.15.0a20260627
...
[O] pip install --index-url https://rocm.nightlies.amd.com/whl-multi-arch/ "torch[device-gfx1151]"==2.12.0+rocm7.15.0a20260626
[O] pip install --index-url https://rocm.nightlies.amd.com/whl-multi-arch/ "torch[device-gfx1151]"==2.12.0+rocm7.15.0a20260626
[O] pip install --index-url https://rocm.nightlies.amd.com/whl-multi-arch/ "torch[device-gfx1151]"==2.12.0+rocm7.14.0a20260615
[O] pip install --index-url https://rocm.nightlies.amd.com/whl-multi-arch/ "torch[device-gfx1151]"==2.12.0+rocm7.14.0a20260531
```

### Minimal  code
head_dim=64/128/256 are all unaffected. Only head_dim=512 triggers the bug.
```
import torch, warnings
from torch.nn.functional import scaled_dot_product_attention
from torch.nn.attention import SDPBackend, sdpa_kernel

torch.manual_seed(0)
q = torch.randn(1, 1, 256, 512, dtype=torch.float16, device='cuda')
k = torch.randn(1, 1, 256, 512, dtype=torch.float16, device='cuda')
v = torch.randn(1, 1, 256, 512, dtype=torch.float16, device='cuda')

with sdpa_kernel(SDPBackend.MATH):
    out_math = scaled_dot_product_attention(q, k, v)

with warnings.catch_warnings():
    warnings.simplefilter('ignore')
    out_default = scaled_dot_product_attention(q, k, v)

diff = (out_default - out_math).abs()
print(f"max_err={diff.max().item():.1f}  output_max={out_default.abs().max().item():.1f}")
# broken:  max_err=512.5  output_max=512.0   <- saturated to head_dim value
# working: max_err=0.000  output_max=0.134
```

### Versions

```
Collecting environment information...
PyTorch version: 2.14.0a0+rocm7.15.0a20260627
Is debug build: False
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: 7.14.60850

OS: Microsoft Windows 11 Pro (10.0.26200 64-bit)
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.13.7 | packaged by conda-forge | (main, Sep  3 2025, 14:18:04) [MSC v.1944 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-11-10.0.26200-SP0
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: 
GPU models and configuration: AMD Radeon(TM) 8060S Graphics (gfx1151)
Nvidia driver version: Could not collect
cuDNN version: Could not collect
Is XPU available: False
HIP runtime version: 7.14.60850
MIOpen runtime version: 3.5.2
Is XNNPACK available: False
Caching allocator config: N/A

CPU:
Name: AMD RYZEN AI MAX+ 395 w/ Radeon 8060S          
Manufacturer: AuthenticAMD
Family: 107
Architecture: 9
ProcessorType: 3
DeviceID: CPU0
CurrentClockSpeed: 3000
MaxClockSpeed: 3000
L2CacheSize: 16384
L2CacheSpeed: None
Revision: 28672

Versions of relevant libraries:
[pip3] amd-torch-device-gfx11==2.12.0+rocm7.15.0a20260626
[pip3] amd-torch-device-gfx1151==2.14.0a0+rocm7.15.0a20260627
[pip3] amd-torch-device-gfx115x==2.14.0a0+rocm7.15.0a20260627
[pip3] numpy==1.26.4
[pip3] torch==2.14.0a0+rocm7.15.0a20260627
[conda] Could not collect
```

cc @peterjc123 @mszhanyi @skyline75489 @nbcsm @iremyux @Blackhex @nkhasbag-nv @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @jataylo @hongxiayang @naromero77amd @pragupta @jerrymannil @xinyazhang @drisspg @liangel-02 @howardzhang-cv

### PR DESCRIPTION
**Pull request by xinyazhang**
The compiled kernel is incorrect due to register/LDS pressures. Disable for now.

This fixes #189849

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @jataylo @hongxiayang @naromero77amd @pragupta @jerrymannil

### CODE REVIEW

File: aten/src/ATen/native/transformers/cuda/sdp_utils.cpp [hunk 1]
```diff
@@ -161,14 +161,14 @@ int64_t minimum_gemm_alignment(sdp_params const& params) {
 #if USE_ROCM_ATTENTION
 inline int aotriton_max_hdim() {
   static const int max_hdim = []() {
-#if AOTRITON_VERSION_CURRENT == AOTRITON_VERSION_INT(0, 11)
+#if AOTRITON_VERSION_CURRENT >= AOTRITON_VERSION_INT(0, 11)
     // gfx11xx only support hdim <= 256 on AOTriton 0.11/0.12
     auto dprops = at::cuda::getCurrentDeviceProperties();
```
[the remark this answers is not in the document]
Comment (xinyazhang) [reply aten/src/ATen/native/transformers/cuda/sdp_utils.cpp:171]: Mixing GPU architectures is not support ATM.

[the remark this answers is not in the document]
Comment (jeffdaily) [reply aten/src/ATen/native/transformers/cuda/sdp_utils.cpp:171]: Agree, not supported atm.

**Review by jeffdaily — APPROVED** [review]
